session.go

```diff
@@ -398,10 +420,17 @@ func Connect(ctx context.Context, opts ...ConnectOption) (Session, error) {
 		remoteUpdateErr = &empty
 	}
 
+	cfg.ClientTypes = append([]string{"library/official/go"}, cfg.ClientTypes...)
+	cfg.ClientVersions = append([]string{libraryAgentVersion}, cfg.ClientVersions...)
+
+	version := strings.Join(cfg.ClientVersions, ",")
+	userAgent := strings.Join(cfg.ClientTypes, ",")
+
 	auth := proto.AuthExtra{
-		Version:            libraryAgentVersion,
+		Version:            version,
 		Authtoken:          proto.ObfuscatedString(cfg.Authtoken),
 		Metadata:           cfg.Metadata,
+		UserAgent:          userAgent,
```

Comment:
This still needs to be the `ClientType` field (down on line 443) instead of the `UserAgent` field to match Rust.

Edit: Wait, we already had a `UserAgent` field? And `ClientType` is actually the newer field?? Why didn't I just use that in the first place? Someone should have stopped me! 😅

Reply:
Wait there's a `ClientType` field? :rofl:

okay i'm going to set that AND set the user agent; opinions on whether the UA follows `name1/version1 name2/version2` or `name1,name2/version1,version2` format?